@here/harp-mapview/lib/PickListener.ts

```diff
@@ -133,9 +148,22 @@ export class PickListener {
     }
 
     private sortResults(): void {
-        if (!this.m_sorted) {
-            this.m_results.sort(defaultSort);
-            this.m_sorted = true;
+        if (this.m_sorted) {
+            return;
         }
+
+        // HARP-14531: group and sort zero-distance results separately and merge them back
+        // to ensure that screen-space objects (e.g. labels) are returned first,
+        // as they are currently rendered on top.
+        const zeroDistanceGroup: PickResult[] = [];
```

Comment:
Why do you need to split & merge?

Why not just change the `defaultSort` function to sort those with zero distance to be higher?

Reply:
I wanted the `defaultSort()` to remain a relatively small comparator function, meaning it applies criteria one by one and compares same properties within a given criterion.

Basically, I didn't want to add `if (lhs.distance === 0 && rhs.distance !== 0)` kind of conditions to this function, thinking that splitting->sorting->merging is expressing the end result better, like "first there are zero-distant items sorted according to 3-step criteria, then the remaining ones, using the same criteria".

Reply:
Ok, I agree, it makes things more complicated.